src/main/java/me/leoko/advancedban/manager/PunishmentManager.java

```diff
@@ -134,6 +134,9 @@ public void discard(String name) {
             try (ResultSet rs = DatabaseManager.get().executeResultStatement(current ? SQLQuery.SELECT_USER_PUNISHMENTS : SQLQuery.SELECT_USER_PUNISHMENTS_HISTORY, target)) {
                 while (rs.next()) {
                     Punishment punishment = getPunishmentFromResultSet(rs);
+                    if(put.getBasic()==PunishmentType.WARNING && ( put == punishment.getType().getBasic() || punishment.getType().getBasic() == PunishmentType.TEMP_WARNING) && (!current || !punishment.isExpired())) {
+                    	  ptList.add(punishment);
+                	}else
```

Comment:
For the warns command it was sending a null for PunishmentType which would work with how bans work, cause you cant see your own active bans, but it doesn't work with the notes. Because it was returning the notes aswell. I figured to set them together, maybe I could set a another section to return the active warnings.

Reply:
Didn't address any of this with my latest commit because I wanted to see what you say about it. My reasons were there. 
Also forgot a null check for put. So history throws a null pointer. But if this isn't the way to go would rather correct it correctly.

Reply:
I am really confused as to why we used 'null' before in the warns command as that should have returned all punishments ^^

I'm still not 100% sure what you are trying to achieve with your code here. Just using 'PunishmentType.WARNING' should return you all the warnings and 'PunishmentType.NOTE' all the notes. Without the need for that additional if-statement.

Also notice how 'put.getBasic()' is the same as 'put' as you are required to use a basic type anyway for the put parameter (see java docs above)
And `punishment.getType().getBasic() == PunishmentType.TEMP_WARNING` always returns false as `TEMP_WARNING` is not the basic type. The basic type is `WARNING`.

Reply:
That was the problem, I was assuming that it would return true on one or the other. That is why I did add that to keep the warns command working. But I should have checked what the getBasic did properly rather than making assumptions, which the getType should have clued me in that I should have checked.

Will revert.